File: app/src/main/java/com/nimtego/plectrum/data/network/AppNetwork.java
package com.nimtego.plectrum.data.network;

import com.nimtego.plectrum.data.network.itunes.ITunesApi;
import com.nimtego.plectrum.data.network.rss_itunes.RssItunesApi;
import com.nimtego.plectrum.data.network.wiki.RestCountries;
import com.nimtego.plectrum.data.network.wiki.WikiApi;

public interface AppNetwork {
    ITunesApi getITunesClient();
    WikiApi getWikiClient(RestCountries restCountries);
    WikiApi getWikiClient();
    RssItunesApi getRssItunesAPi();
}

File: app/src/main/java/com/nimtego/plectrum/data/network/AppNetwork.kt
package com.nimtego.plectrum.data.network

import com.nimtego.plectrum.data.network.itunes.ITunesApi
import com.nimtego.plectrum.data.network.rss_itunes.RssItunesApi
import com.nimtego.plectrum.data.network.wiki.RestCountries
import com.nimtego.plectrum.data.network.wiki.WikiApi

interface AppNetwork {
    val iTunesClient: ITunesApi
    val wikiClient: WikiApi
    val rssItunesAPi: RssItunesApi
    fun getWikiClient(restCountries: RestCountries): WikiApi
}
